Adds anonymous pulls procedure for image registry (#76038)
openshift/openshift-docs: modules/enabling-anonymous-pulls.adoc

// Module included in the following assemblies:
//
// * registry/accessing-the-registry.adoc

:_mod-docs-content-type: CONCEPT
[id="enabling-anonymous-pulls_{context}"]
= Enabling anonymous pulls

> The following procedure shows you how to check if your cluster has the proper roles and permissions for anonymous pulls and, if it does not, how to enable anonymous requests to the self access reviewer source.

Reviewer: This sentence mixes up the enabling of anonymous pulls with anonymous requests to the self access reviewer resource - we should be consistent.
```suggestion
The following procedure shows you how to check if your cluster has the proper roles and permissions for anonymous pulls and, if it does not, how to enable anonymous pulls.
```

.Procedure 

. Enter the following command to see if the `system:unauthenticated` group is included in the `subjects` parameter of your `ClusterRoleBinding` object:
+
[source,terminal]
----
$ oc get clusterrolebindings self-access-reviewers -o yaml
----
+
.Example output
+
[source,terminal]
----
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  annotations:
    rbac.authorization.kubernetes.io/autoupdate: "true"
  creationTimestamp: "2024-01-17T20:50:41Z"
  name: self-access-reviewers
  resourceVersion: "14477"
  uid: d0ae964c-89ce-4b24-8905-a2318484fdd5
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: self-access-reviewer
subjects:
- apiGroup: rbac.authorization.k8s.io
  kind: Group
  name: system:authenticated
----

> . If your cluster has the `self-access-reviewer` cluster role binding available to the `system:unauthenticated` group, you do not need to create a new cluster role binding, and can move on to the following step.

Reviewer 2: Maybe
```suggestion
. If your cluster has the cluster role binding that `self-access-reviewer` cluster role is binded to the `system:unauthenticated` group, you do not need to create a new cluster role binding, and can move on to the following step.
```

+

> If `self-access-reviewer` is not available to the `system:unauthenticated` group, or the cluster role binding does not exist, you can create a file named `add-self-access-review-unauth.yaml` and apply it with the following command:

Reviewer 2: Maybe
```suggestion
If `self-access-reviewer` is not available to the `system:unauthenticated` group, you can create a file named `add-self-access-review-unauth.yaml` and apply it with the following command:
```

+
[source,yaml]
----
cat <<EOF | oc apply -f -
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  annotations:
    rbac.authorization.kubernetes.io/autoupdate: "true"
  name: restore-self-access-reviewers-unauth
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: self-access-reviewer
subjects:
- apiGroup: rbac.authorization.k8s.io
  kind: Group
  name: system:unauthenticated
EOF
----
+
.Example output
+
[source,terminal]
----
clusterrolebinding.rbac.authorization.k8s.io/restore-self-access-reviewers-unauth created
----

> . Enter the following command to grant the `system:image-puller` role to the `system:unauthenticated` group in the specified namespace, enabling anonymous users to pull images:

Reviewer 2: How about add openshift namespace example, to keep consistent with step in verification.
```suggestion
. Enter the following command to grant the `system:image-puller` role to the `system:unauthenticated` group in the specified namespace,such as openshift namespcae, enabling anonymous users to pull images:
```

+
[source,terminal]
----

> $ oc adm policy add-role-to-group system:image-puller system:unauthenticated --namespace <namespace>

Reviewer 2: ```suggestion
$ oc adm policy add-role-to-group system:image-puller system:unauthenticated --namespace openshift
```

// Module included in the following assemblies:
//
// * registry/accessing-the-registry.adoc

:_mod-docs-content-type: PROCEDURE
[id="enabling-anonymous-pulls_{context}"]
= Enabling anonymous pulls

> In some cases, administrators might want to enable anonymous pulls from the image registry, for example, when using CI/CD pipelines to streamline access to base images, or reduce authentication overhead. 

Reviewer 3: ```suggestion
In some cases, administrators might want to enable anonymous pulls from the image registry, for example, when using CI/CD pipelines to streamline access to base images or reduce authentication overhead. 
```

Previously, enabling anonymous pulls involved configuring the internal image registry to accept such requests by running the following command:

[source,terminal]
----
$ oc adm policy add-role-to-group system:image-puller system:unauthenticated --namespace openshift
----

> With {product-title} {product-version}, changes made to the API server and the `selfsubjectaccessreviews` resource now require administrators to add unauthenticated groups to the `self-access-review` Cluster Role. This is only required for new clusters installed from {product-title} {product-version}; clusters that are upgrading to {product-version} from a previous version can continue using the `oc adm policy add-role-to-group system:image-puller system:unauthenticated --namespace openshift` command. 

Reviewer 3: ```suggestion
With {product-title} {product-version}, changes to the API server and the `selfsubjectaccessreviews` resource now require administrators to add unauthenticated groups to the `self-access-review` Cluster Role. This is only required for new clusters installed from {product-title} {product-version}; clusters that are upgrading to {product-version} from a previous version can continue using the `oc adm policy add-role-to-group system:image-puller system:unauthenticated --namespace openshift` command. 
```

> Depending on your {product-title} version, and whether you installed a new cluster from version 4.16 or later or upgraded to one of those versions from an earlier release, you can use one of the following commands to enable anonymous pulls from the internal image registry.

Reviewer 3: ```suggestion
Depending on your {product-title} version and whether you installed a new cluster from version 4.16 or later or upgraded to one of those versions from an earlier release, you can use one of the following commands to enable anonymous pulls from the internal image registry.
```
